Seeing this error now in the tests after merging the other PR which fixed circle:

```
2020-11-30 18:14:32,468Z    tap - Traceback (most recent call last):
2020-11-30 18:14:32,468Z    tap -   File "/usr/local/share/virtualenvs/tap-square/bin/tap-square", line 33, in <module>
2020-11-30 18:14:32,468Z    tap -     sys.exit(load_entry_point('tap-square==1.0.1', 'console_scripts', 'tap-square')())
2020-11-30 18:14:32,468Z    tap -   File "/usr/local/share/virtualenvs/tap-square/lib/python3.5/site-packages/singer/utils.py", line 229, in wrapped
2020-11-30 18:14:32,468Z    tap -     return fnc(*args, **kwargs)
2020-11-30 18:14:32,468Z    tap -   File "/usr/local/share/virtualenvs/tap-square/lib/python3.5/site-packages/tap_square/__init__.py", line 23, in main
2020-11-30 18:14:32,468Z    tap -     sync(args.config, args.state, catalog)
2020-11-30 18:14:32,468Z    tap -   File "/usr/local/share/virtualenvs/tap-square/lib/python3.5/site-packages/tap_square/sync.py", line 99, in sync
2020-11-30 18:14:32,468Z    tap -     state = singer.write_bookmark(state, tap_stream_id, replication_key, max_record_value)
2020-11-30 18:14:32,468Z    tap - UnboundLocalError: local variable 'max_record_value' referenced before assignment
```
My assumption is that this was tested for incremental streams but maybe for non-incremental streams this write bookmark code path needs to either set the max_record_value or it needs to not try to write the bookmark?

--- a/tap_square/sync.py
+++ b/tap_square/sync.py
@@ -63,6 +63,7 @@ def sync(config, state, catalog):
                     singer.write_state(state)
 
                 state = singer.clear_bookmark(state, tap_stream_id, 'sync_start')
+                state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
                 state = singer.write_bookmark(
                     state,
                     tap_stream_id,
@@ -98,8 +99,8 @@ def sync(config, state, catalog):
                         transformed_record,
                     )
 
-            state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
-            singer.write_state(state)
+                state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
+                singer.write_state(state)
 
     state = singer.set_currently_syncing(state, None)
     singer.write_state(state)

--- a/tap_square/sync.py
+++ b/tap_square/sync.py
@@ -87,6 +87,9 @@ def sync(config, state, catalog):
                     state = singer.write_bookmark(state, tap_stream_id, 'cursor', cursor)
                     singer.write_state(state)
 
+                state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
+                state = singer.write_bookmark(state, tap_stream_id, replication_key, max_record_value)
+                singer.write_state(state)
             else:
                 for record in stream_obj.sync(start_time, bookmarked_cursor):
                     transformed_record = transformer.transform(record, stream_schema, stream_metadata)
@@ -95,9 +98,6 @@ def sync(config, state, catalog):
                         transformed_record,
                     )
 
-            state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
-            state = singer.write_bookmark(state, tap_stream_id, replication_key, max_record_value)
-            singer.write_state(state)
 
     state = singer.set_currently_syncing(state, None)
     singer.write_state(state)

--- a/tap_square/sync.py
+++ b/tap_square/sync.py
@@ -98,6 +98,8 @@ def sync(config, state, catalog):
                         transformed_record,
                     )
 
+            state = singer.clear_bookmark(state, tap_stream_id, 'cursor')
+            singer.write_state(state)
 
     state = singer.set_currently_syncing(state, None)
     singer.write_state(state)

